consoles/console.pm

```diff
@@ -74,10 +74,15 @@ sub screen {
 }
 
 # helper function
+# Keep ssh session for the maximum of ServerAliveCountMax x ServerAliveInterval seconds 
+# even without receiving any messsage back from the server, and this will not affect normal
+# ssh disconnect and console switching. Ssh console may not display returned result of 
+# long-time run test without these options. TCPKeepAlive ensures that if network goes down 
+# or the remote host dies, machines will be properly noticed
 sub sshCommand {
     my ($self, $username, $host, $gui) = @_;
 
-    my $sshopts = "-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PubkeyAuthentication=no $username\@$host";
+    my $sshopts = "-o TCPKeepAlive=yes -o ServerAliveCountMax=480 -o ServerAliveInterval=60 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PubkeyAuthentication=no $username\@$host";
```

Comment:
better extract the values you provide into variables and just define default values to prevent having "magic numbers" here. E.g.

```perl
my $server_alive_count_max = get_var('_SSH_SERVER_ALIVE_COUNT_MAX', 480);
my $server_alive_interval = get_var('_SSH_SERVER_ALIVE_INTERVAL', 60);
my $sshopts = "-o TCPKeepAlive=yes -o ServerAliveCountMax=$server_alive_count_max -o ServerAliveInterval=$server_alive_interval -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PubkeyAuthentication=no $username\@$host";
```

and describe the variables in doc/backend_vars.asciidoc

I realized the method "sshCommand" is only used within two specific console files and should not be defined in the generic base level file "consoles/console.pm". I will handle this in a separate PR. This will cause a conflict with your code changes but I can handle that after merging your PR then.

Reply:
Why there is preceding underscore to these variables ?

Reply:
This is commonly done for test parameters that are not to be used within the test distribution itself. You can also leave out the leading underscore if you prefer

Reply:
Done. underscore is kept.